app/code/Magento/PageCache/Observer/FlushCacheByTags.php

```diff
@@ -21,78 +30,59 @@ class FlushCacheByTags implements ObserverInterface
     /**
      * Application config object
      *
-     * @var \Magento\PageCache\Model\Config
+     * @var Config
      */
     protected $_config;
 
     /**
-     * @var \Magento\PageCache\Model\Cache\Type
+     * @var Type
      */
     private $fullPageCache;
 
     /**
      * Invalidation tags resolver
      *
-     * @var \Magento\Framework\App\Cache\Tag\Resolver
+     * @var Resolver
      */
     private $tagResolver;
 
     /**
-     * @param \Magento\PageCache\Model\Config $config
-     * @param \Magento\Framework\App\PageCache\Cache $cache
+     * @param Config $config
+     * @param Cache $cache
+     * @param Type $fullPageCache
+     * @param Resolver $tagResolver
      */
-    public function __construct(\Magento\PageCache\Model\Config $config, \Magento\Framework\App\PageCache\Cache $cache)
-    {
+    public function __construct(
+        Config $config,
+        Cache $cache,
+        Type $fullPageCache,
+        Resolver $tagResolver
+    ) {
         $this->_config = $config;
         $this->_cache = $cache;
+        $this->fullPageCache = $fullPageCache;
```

Comment:
Hi @rogyar, I asked similar question in one of my previous PRs, such non-backward compatible changes I'm doing in those PRs are based on info I received in https://github.com/magento/magento2/pull/26269, where I was asked to replace backward-compatible change with one that doesn't use ObjectManager, since we're developing for 2.4 now.
Also such changes are being directly merged to core by Magento core team https://github.com/magento/magento2/commit/4e6f6ae42cc6109a02d35d0363b1cad0bbac66e1

Reply:
Hi @Bartlomiejsz. You are right, that's fair since we are editing a non-API class. That's rather a professional habit avoid bringing backward-incompatible changes :)